Gifski/PreviewGenerator.swift

```diff
@@ -0,0 +1,220 @@
+//
+//  PreviewGenerator.swift
+//  Gifski
+//
+//  Created by Michael Mulet on 3/22/25.
+
+import Foundation
+import SwiftUI
+import AVFoundation
+
+@Observable
+final class PreviewGenerator {
+	@MainActor
+	var previewImage: PreviewImage?
+
+	private var previewImageCommand: PreviewCommand?
+	@MainActor
+	var imageBeingGeneratedNow = false
+
+	private var commandStream = LatestCommandAsyncStream()
+
+	enum PreviewImage {
+		case image(NSImage)
+		case tooFewFrames
+	}
+
+	init() {
+		Task(priority: .utility) {
+			for await item in commandStream {
+				guard commandStream.commandIsLatest(command: item) else {
+					continue
+				}
+				if previewImageCommand == item.command {
+					/**
+					 Don't regenerate if it's the same
+					 */
+					continue
+				}
+				Task {
```

Comment:
Task is not synchronous, so you are introducing race issues.

Use `await MainActor.run {}` instead.